ui/apps/platform/src/services/VulnerabilityExceptionService.ts

```diff
@@ -0,0 +1,94 @@
+import { generatePath } from 'react-router-dom';
+
+import { SlimUser } from 'types/user.proto';
+import { SearchFilter, ApiSortOption } from 'types/search';
+import { getListQueryParams } from 'utils/searchUtils';
+import { Empty } from './types';
+import axios from './instance';
+
+const baseUrl = '/v2/vulnerability-exceptions';
+
+export type VulnerabilityState = 'OBSERVED' | 'DEFERRED' | 'FALSE_POSITIVE';
+
+export type ExceptionStatus = 'PENDING' | 'APPROVED' | 'DENIED' | 'APPROVED_PENDING_UPDATE';
+
+type Comment = {
+    id: string;
+    message: string;
+    user: SlimUser;
+    createdAt: string; // ISO 8601
+};
+
+type Scope = {
+    imageScope: {
+        registry: string;
+        remote: string;
+        tag: string;
+    };
+};
+
+type ExceptionExpiry =
+    | {
+          expiryType: 'TIME';
+          expiresOn: string | null;
+      }
+    | {
+          expiryType: 'ALL_CVE_FIXABLE' | 'ANY_CVE_FIXABLE';
+      };
+
+export type VulnerabilityException = {
+    id: string;
+    name: string;
+    targetState: VulnerabilityState;
+    exceptionStatus: ExceptionStatus;
+    expired: boolean;
+    requester: SlimUser;
+    createdAt: string; // ISO 8601
+    lastUpdated: string; // ISO 8601
+    comments: Comment[];
+    scope: Scope;
+    deferralReq?: {
+        expiry: ExceptionExpiry;
+    };
+    fpRequest?: Empty;
```

Comment:
Interesting TypeScript question, not a blocker to merge.
* If it is really either-or, it is worth expressing that for any conditional code in components?
    https://github.com/stackrox/stackrox/blob/master/proto/api/v2/vuln_exception_service.proto#L89-L92
* While it is fresh in our minds, and the minds of backend teammates, is there any interdependency:
    https://github.com/stackrox/stackrox/blob/master/proto/api/v2/vuln_exception_service.proto#L97-L100

Reply:
Oh good call, yes these are mutually exclusive and the request type and update type are in fact interdependent.

I prefer to have these types well defined when possible, so I'll look into converting these to more accurate representations via a union.